tests/unit/player/directives/track.js

```diff
@@ -1,9 +1,21 @@
 "use strict";
 
 describe("FM.player.trackDirective", function() {
-    var element, $scope, $rootScope, isolatedScope, $templateCache;
-
-    beforeEach(module("FM.player.trackDirective"));
+    var element, $scope, $rootScope, isolatedScope, $templateCache, PlayerQueueResource;
+
+    beforeEach(function(){
+        module("FM.player.trackDirective")
+        module(function ($provide) {
+            $provide.provider("PlayerQueueResource", function () {
+                this.$get = function () {
+                    PlayerQueueResource = {
+                        save: jasmine.createSpy("PlayerQueueResource.save")
+                    }
+                    return PlayerQueueResource;
+                }
+            })
+        })
+    });
```

Comment:
@jamesjwarren instead of mocking the resources methods, a better way would be to just mock the api response using [$httpBackend](https://docs.angularjs.org/api/ngMock/service/$httpBackend)

example:

```javascript
$httpBackend
    .when(API_URL + "/queue")
    .expect(200, ["example queue data"])
```